src/main/java/crazypants/enderio/conduit/gui/GuiExternalConnectionSelector.java

```diff
@@ -111,6 +125,14 @@ public void drawScreen(int par1, int par2, float par3) {
     int y = height / 2 - butHeight * 3 - 5;
         
     drawString(Minecraft.getMinecraft().fontRendererObj, txt, x, y, ColorUtil.getARGB(Color.white));
+    for (int i = 0; i < 6; i++) {
+      String blockName = adjacentBlockNames[i];
+      if (blockName == null) {
+        continue;
+      }
+      int textWidth = Minecraft.getMinecraft().fontRendererObj.getStringWidth(blockName);
+      drawString(Minecraft.getMinecraft().fontRendererObj, blockName, textPositions[i].x + 60 / 2 - textWidth / 2, textPositions[i].y, ColorUtil.getARGB(Color.gray));
```

Comment:
Can you move this up, so it is rendered before ("behind") the buttons? There is a chance of overlap here.